lib/media/media_source_engine.js

```diff
@@ -584,40 +598,112 @@ shaka.media.MediaSourceEngine = class {
       return;
     }
 
+    const needTimestampOffsetCalculation = !this.sequenceMode_ &&
```

Comment:
I have renamed the variable. I don't want it to return an error because I've seen cases where the content is bad, but doing this works correctly, because not in all segments I need to update timestampOffset.